frappe/utils/file_manager.py

```diff
@@ -147,36 +149,43 @@ def save_file(fname, content, dt, dn, folder=None, decode=False, is_private=0):
 			content = content.split(",")[1]
 		content = base64.b64decode(content)
 
-	file_size = check_max_file_size(content)
-	content_hash = get_content_hash(content)
-	content_type = mimetypes.guess_type(fname)[0]
-	fname = get_file_name(fname, content_hash[-6:])
-	file_data = get_file_data_from_hash(content_hash, is_private=is_private)
-	if not file_data:
-		call_hook_method("before_write_file", file_size=file_size)
-
-		write_file_method = get_hook_method('write_file', fallback=save_file_on_filesystem)
-		file_data = write_file_method(fname, content, content_type=content_type, is_private=is_private)
-		file_data = copy(file_data)
-
-	file_data.update({
-		"doctype": "File",
-		"attached_to_doctype": dt,
-		"attached_to_name": dn,
-		"folder": folder,
-		"file_size": file_size,
-		"content_hash": content_hash,
-		"is_private": is_private
-	})
-
-	f = frappe.get_doc(file_data)
-	f.flags.ignore_permissions = True
-	try:
-		f.insert()
-	except frappe.DuplicateEntryError:
-		return frappe.get_doc("File", f.duplicate_entry)
-
-	return f
+	if '.' in fname and 'zip' == fname.split('.')[-1]:
+		zip_data = StringIO()
+		zip_data.write(content)
+		file_path = get_files_path(is_private=is_private)
+		with zipfile.ZipFile(zip_data) as z:
+			z.extractall(file_path)
+	else:
```

Comment:
I didn't create **File** for extracted files according to requirements for this task created by Jerome (https://discuss.erpnext.com/users/jerome76/activity). He write "File metadata should NOT be inserted into the database, because the file could be a "Thumbnail URL" - image which should not be listed as a separate file in the "File Manager". Let's discuss with him and if will decided to create ***File** for extractted files I ill do it